src/tools/compiletest/src/runtest.rs

```diff
@@ -2004,6 +2004,16 @@ impl<'test> TestCx<'test> {
             Some(CompareMode::Chalk) => {
                 rustc.args(&["-Zchalk"]);
             }
+            Some(CompareMode::SplitDwarf) => {
+                rustc.args(&["-Zsplit-dwarf=split"]);
+            }
+            Some(CompareMode::SplitDwarfSingle) => {
+                // FIXME(davidtwco): Single mode keeps debuginfo in the same object file, but in
+                // such a way that it it skipped by the linker - so it's expected that when codegen
+                // units are linked together that this debuginfo would be lost without keeping
+                // around the temps.
+                rustc.args(&["-Zsplit-dwarf=single", "-Csave-temps"]);
```

Comment:
I think `preserve_objects_for_their_debuginfo` should return true when using `-Zsplit-dwarf=single`. That would make the `-Csave-temps` unnecessary.

Reply:
Good idea, fixed!